client-v2/src/shared/selectors/shared.ts

```diff
@@ -172,8 +172,35 @@ const createCollectionStageGroupsSelector = () => {
       collection: Collection | void,
       groups: { [id: string]: Group },
       stage: CollectionItemSets
-    ): Group[] =>
-      ((collection && collection[stage]) || []).map(id => groups[id])
+    ): Group[] => {
+      const grps = ((collection && collection[stage]) || []).map(
+        id => groups[id]
+      );
+      if (grps.length < 2) {
+        return grps;
+      }
+
+      // Groups without names and ids are groups which no longer exist in the config because
+      // the collection layout has changed. We need to collect the article fragments in these
+      // groups and display them in the top group.
+      const orphanedFragments: string[] = grps
+        .filter(grp => !grp.name && grp.id)
+        .reduce((frags: string[], g) => frags.concat(g.articleFragments), []);
+
+      // The final array of groups consist of groups where all groups without names but with ids
+      // are filtered out as these groups no longer exist in the config of the collection.
+      const finalGroups = grps.filter(grp => grp.name || !grp.id);
+      if (finalGroups.length > 0) {
+        const grpF = finalGroups ? finalGroups[0].articleFragments : [];
```

Comment:
We already know that `finalGroups` exists and has a length -- is this ternary necessary?

Reply:
(also just to satisfy my own curiosity does grpF mean first group? 🤓 )

Reply:
Variables really shouldn't be called grpF - I have renamed to `originalFirstGroupArticleFragments` to be more descriptive.